docs/ref/csrf.txt

```diff
@@ -49,27 +49,32 @@ To take advantage of CSRF protection in your views, follow these steps:
    ``'django.template.context_processors.csrf'`` context processor is
    being used. Usually, this can be done in one of two ways:
 
-   1. Use RequestContext, which always uses
+   1. Use ``RequestContext``, which always uses
       ``'django.template.context_processors.csrf'`` (no matter what template
       context processors are configured in the :setting:`TEMPLATES` setting).
-      If you are using generic views or contrib apps, you are covered already,
-      since these apps use RequestContext throughout.
+      If you are using the :func:`django.shortcuts.render` function, generic
+      views, or contrib apps, you are covered already, since these all use
+      ``RequestContext``.
 
-   2. Manually import and use the processor to generate the CSRF token and
-      add it to the template context. e.g.::
+   2. XXX -- Remove??
```

Comment:
I was looking at this section of the docs as part of [#25854](https://code.djangoproject.com/ticket/25854). (still a WIP I'm afraid).

I see many new users encountering CSRF errors, who are using `c.update(csrf(request))`. I always wondered why they were doing that instead of just using `render`. I think it might be because of the snippet below.

Firstly, mentioning `render` above is a good change, I had written something similar.

Next, my thought was to add a snippet so that users saw the render example first.

    def my_view(request):
        c = {}
        ...
        return render(request, 'a_template.html', c)

But I think your suggestion to simply delete section 2. would be better.